tests/test_peer_heap.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+import sys
+
+import mock
+import random
+import pytest
+from tchannel.peer_heap import PeerHeap
+
+
+n = 1000
+
+
+@pytest.fixture
+def peer_heap():
+    return PeerHeap()
+
+
+def mock_peer(score=None):
+    peer = mock.MagicMock()
+    peer.index = -1
+    peer.score = score if score is not None else random.randint(0, sys.maxint)
+    return peer
+
+
+def mock_peers(n):
+    peers = []
+    for i in range(n):
+        peers.append(mock_peer(i))
+
+    random.shuffle(peers)
+    return peers
+
+
+def verify(ph, parent):
+    child1 = 2*parent + 1
+    child2 = 2*parent + 2
+    if child2 < ph.size():
+        assert ph.lt(parent, child1)
+        verify(ph, child1)
+
+    if child2 < ph.size():
+        assert ph.lt(parent, child2)
+        verify(ph, child2)
+
+
+def test_push(peer_heap):
```

Comment:
I wasn't asking to duplicate these tests. test_heap.py is enough to test the heap implementation itself.

For this file, your assumption should be that the heap implementation works as expected. You only need to do a sort of sanity check that PeerHeap's own operations work, and that it does what we say HeapOperation does.

Like, make sure that `pop_peer()` doesn't explode when the list is empty. Stuff like that. Basically, test_heap gives us confidence in the heap implementation and test_peer_heap gives us confidence that PeerHeap knows how to work with that heap implementation without blowing up.